File: tests/js/test_verify_email.js
var assert = require('assert');
var createSession = require('./utils/session.js');

describe('verify email', function() {
    beforeEach(function(done) {
        browser
            .url('http://localhost:8537')
            .setCookie({
                name: 'session',
                value: createSession('alice')
            })
            .call(done);
    });

    afterEach(function(done) {
        browser
            .url('http://localhost:8537')
            .deleteCookie('session')
            .call(done);
    });

    it('should not throw error on clicking resend', function(done) {
        browser
            .url('http://localhost:8537/~alice/emails/verify.html?email=alice@gratipay.com&nonce=abcd')
            .getText('#content p:first-of-type', function(err, text) {
                assert.equal(text, 'The verification code for alice@gratipay.com is bad.');
            })
            .click("button.resend")
            .waitForExist('.notification.notification-success', 2000).then(function(isExisting) {
                assert(isExisting, 'Notification element not appended!');
            })
            .getText('.notification.notification-success', function(err, text) {
                var didSend = (text[1].indexOf('A verification email has been sent') > -1);
                assert(didSend, 'Expected email to be sent');
            })
            .call(done);
    });
});

File: tests/ttw/test_verify_email.py
from __future__ import absolute_import, division, print_function, unicode_literals

from gratipay.testing import BrowserHarness


class Tests(BrowserHarness):

    def test_clicking_resend_doesnt_error(self):
        self.make_participant('alice', claimed_time='now')
        self.sign_in('alice')
        self.visit('/~alice/emails/verify.html?email=alice@gratipay.com&nonce=abcd')
        assert self.has_text('The verification code for alice@gratipay.com is bad.')
        self.css('button.resend').click()
        assert self.has_element('.notification.notification-success', 2)
        assert self.has_text('A verification email has been sent')
